## **STAR alignment**

After indexing the reads, the next step is to align them, which leads to the subsequent .sh and .sge scripts:

```{bash .sge, eval=FALSE}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the script since this line
#
# Your job name
$ -N fastqc_C2_T2
#
# Send an email after the job has finished
$ -m e
$ -M paurodriguez760@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next lin$
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
#
module load star/2.7.9a
cd /mnt/Guanina/bioinfo24/Equipo_rosa/scripts/
./alignment_STAR.sh
```

```{bash .sh, eval=FALSE}

#ruta en donde est√° el index de referencia
index=/mnt/Guanina/bioinfo24/Equipo_rosa/STAR_index

#ruta de los archivos de secuencias
FILES=/mnt/Guanina/bioinfo24/Equipo_rosa/data_trimmed/*_1_trimmed.fq

for f in $FILES
do
    base=$(basename $f _1_trimmed.fq)
    echo $base
    STAR --runThreadN 12 --genomeDir $index --readFilesIn $f /mnt/Guanina/bioinfo24/Equipo_rosa/data_trimmed/$base"_2_trimmed.fq" \
    --outSAMtype BAM SortedByCoordinate \
    --quantMode GeneCounts \
    --outFileNamePrefix /mnt/Guanina/bioinfo24/Equipo_rosa/STAR_output/$base
done
```


**Where:**

`for f in $FILES:` traverses each sequence file in variable FILES.

`base=$(basename $f _1_trimmed.fq):` extracts the base file name by removing the _1_trimmed.fq extension. This base name will be used later to name the output files.

`--runThreadN 12:` indicates that 12 processing threads should be used.

`--genomeDir $index:` specifies the path of the reference index.

`--readFilesIn $f /mnt/Guanina/bioinfo24/Equipo_rosa/data_trimmed/$base"_2_trimmed.fq" \:` specifies the input read files for STAR. 
`$f` represents read file 1, and /mnt/Guanina/bioinfo24/Equipo_rosa/data_trimmed/$base"_2_trimmed.fq" represents read file 2 corresponding to the same sample pair.

`--outSAMtype BAM SortedByCoordinate:` specifies the output format of alignment files as BAM files sorted by coordinates.

`--quantMode GeneCounts:`  performs quantification of gene expression and generates gene counts.

`--outFileNamePrefix /mnt/Guanina/bioinfo24/Equipo_rosa/STAR_output/$base:` specifies the location and prefix of the output files generated by STAR.


`done:` marks the end of the for loop.


Now, what **output** did the alignment process give us?

![](https://github.com/arianaresi/RNA-seq-Project/blob/main/Alignment/Images/output_STAR_alignment.png)


Now we will see **what each file is for**

**Aligned.sortedByCoord.out.bam:** contains the reads aligned against the reference genome, sorted by genomic coordinates. 

**Log.final.out:** provides an overall summary of the alignment process and the parameters used.

**Log.out:** records events and messages during the execution of the alignment process.

**Log.progress.out:** show the progress of the alignment process.

**ReadsPerGene.out.tab:** contains counts of aligned reads per gene. They are useful for differential gene expression analysis.

**SJ.out.tab:**contains information about splice junctions found during alignment. These junctions are important for identifying introns and alternative splicing events.

**STARtmp:** temporary directories used during the alignment process.


The file we used (and therefore give higher priority to) are the .out.tab files. 

You can open them with a `less file.out.tab`


Here, we explain the .out.tab files:


Each row in this file represents an entry for a specific gene and provides information on the number of reads aligned to that gene.

**Column 1 (chr):** the name of the chromosome or reference sequence where the gene is located.

**Column 2 (start):** the starting position of the gene on the chromosome.

**Column 3 (end):** the end position of the gene on the chromosome.

**Column 4 (strand):** the orientation of the gene (1 for positive sense, -1 for negative sense).

**Column 5 (Length):** the length of the gene.

**Column 6 (EffectiveLength):** the effective length of the gene.

**Column 7 (expected_count):** the expected number of reads aligned to this gene.

**Column 8 (TPM):** transcripts Per Million, a measure of the relative abundance of the gene in the sample.

**Column 9 (FPKM):** fragments Per Kilobase of transcript per Million mapped reads, another measure of the relative abundance of the gene.
